sds/src/match_validation/config.rs

```diff
@@ -74,9 +74,28 @@ pub struct HttpValidatorConfig {
 }
 
 impl HttpValidatorConfig {
-    pub fn new(endpoint: &str) -> Self {
-        HttpValidatorConfig {
-            endpoint: endpoint.to_string(),
+    pub fn new(endpoint: &str, hosts: Vec<String>) -> Result<Self, String> {
```

Comment:
As there is a builder now, I think it is better is to force everyone to use it. TO do so, we can make this method private. This will require to move the `HttpValidatorConfigBuilder` in this class, but I think it's for the best as the builder will be closer to the actual object built (`HttpValidatorConfig`)
```suggestion
   fn new(endpoint: &str, hosts: Vec<String>) -> Result<Self, String> {
```